packages/mui-base/src/utils/evaluateRenderProp.ts

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import type { BaseUIComponentProps } from './BaseUI.types';
+
+export function evaluateRenderProp<ElementType extends React.ElementType, OwnerState>(
+  render: BaseUIComponentProps<ElementType, OwnerState>['render'],
+  props: React.HTMLAttributes<any>,
+  ownerState: OwnerState,
+) {
+  return typeof render === 'function'
+    ? render(props, ownerState)
+    : React.cloneElement(render, { ...props, ...render.props });
```

Comment:
IMO we should use similar prop merging logic we have in the hooks: chaining event handlers and merging styles. I don't think we'll be able to support `preventBaseUIHandler` this way, though.

Reply:
True - `mergeReactProps(render.props, props)` should work fine though?

Reply:
Yup, I think so.